Widgets/Widgets.swift

```diff
@@ -0,0 +1,154 @@
+//
+//  Widgets.swift
+//  Widgets
+//
+//  Copyright © 2020 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import WidgetKit
+import SwiftUI
+import Core
+import Kingfisher
+
+struct Favorite {
+
+    static let ddgDomain = "duckduckgo.com"
+
+    let url: URL
+    let domain: String
+    let favicon: UIImage?
+
+    var isDuckDuckGo: Bool {
+        return domain == Self.ddgDomain
+    }
+
+    var needsColorBackground: Bool {
+        return favicon == nil && domain != Self.ddgDomain
+    }
+
+}
+
+struct Provider: TimelineProvider {
+
+    typealias Entry = FavoritesEntry
+
+    func getSnapshot(in context: Context, completion: @escaping (FavoritesEntry) -> Void) {
+        completion(createEntry(in: context))
+    }
+
+    func placeholder(in context: Context) -> FavoritesEntry {
+        return createEntry(in: context)
+    }
+
+    func getTimeline(in context: Context, completion: @escaping (Timeline<FavoritesEntry>) -> Void) {
+        let timeline = Timeline(entries: [createEntry(in: context)], policy: .atEnd)
```

Comment:
My `WidgetKit` experience is still pretty limited, but since we're refreshing the timeline manually after the favourites change it should be safe to use `.never` for the reload policy.

```suggestion
        let timeline = Timeline(entries: [createEntry(in: context)], policy: .never)
```